core/src/main/java/tc/oc/pgm/community/events/PlayerVanishEvent.java

```diff
@@ -0,0 +1,30 @@
+package tc.oc.pgm.community.events;
+
+import org.bukkit.event.HandlerList;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.event.MatchPlayerEvent;
+
+public class PlayerVanishEvent extends MatchPlayerEvent {
+
+  private final boolean vanished;
```

Comment:
Since this is sorted in the `MatchPlayer` object, you don't need to store it in the event as well.